.tekton/check-no-quay-io-references.yaml

```
@@ -0,0 +1,40 @@
+apiVersion: tekton.dev/v1
+kind: Task
+metadata:
+  name: check-no-quay-io-references
+  namespace: rh-acs-tenant
+spec:
+  description: This task is used to ensure that no quay.io references are present in the catalog files,
+    as they are not allowed in the production environment.
+  params:
+  - name: SOURCE_ARTIFACT
+    description: The Trusted Artifact URI pointing to the artifact with
+      the application source code. This should be the result of the git-clone task,
+      results from other tasks might fail as dirty.
+    type: string
+  volumes:
+  - name: workdir
+    emptyDir: { }
+  - name: deps
+    emptyDir: { }
+  stepTemplate:
+    volumeMounts:
+    - mountPath: /var/workdir
+      name: workdir
+  steps:
+  - name: use-trusted-artifact
+    image: quay.io/redhat-appstudio/build-trusted-artifacts:latest@sha256:9b180776a41d9a22a1c51539f1647c60defbbd55b44bbebdd4130e33512d8b0d
+    args:
+    - use
+    - $(params.SOURCE_ARTIFACT)=/var/workdir/source
+  - name: check-no-quay-io-references
+    image: registry.access.redhat.com/ubi9:latest@sha256:304b50df1ea4db9706d8a30f4bbf26f582936ebc80c7e075c72ff2af99292a54
+    workingDir: /var/workdir/source
+    script: |
+      #!/usr/bin/env bash
+      set -euo pipefail
+
+      if grep -F 'quay.io' catalog-template.yaml catalog-bundle-object/rhacs-operator/catalog.json catalog-csv-metadata/rhacs-operator/catalog.json; then
+        echo >&2 "Catalog file contains references to quay.io. It is not allowed to merge quay.io based bundle images to master branch."
```

Thinking more about this now, what if `quay.io` is mentioned in comments in `catalog-template.yaml`? Well maybe let's have that case first before complicating the check.

But `catalog-csv-metadata/rhacs-operator/catalog.json`. If you check the contents, you'd see a lot of text including CRD descriptions. It could be that at some future version we extend CRDs and mention "quay.io" in description of one of the attributes. This check will start failing.

I suggest trust our other check that both `**/catalog.json` files are derived from `catalog-template.yaml` and here validated only `catalog-template.yaml`.

```suggestion
      if grep -F 'quay.io' catalog-template.yaml; then
        echo >&2 "catalog-template.yaml mentions quay.io. It is not allowed to merge quay.io based bundle images to master branch because it will no longer be releasable."
```